src/entities/amm/amm.ts

```diff
@@ -1482,6 +1483,36 @@ export class AMM {
     return tickToFixedRate(closestUsableTick).toNumber();
   }
 
+  // historocal rates
+
+  public async getHistoricalVariableRate(
+    subgraphUrl: string,
+    filters: {
+      granularity: Granularity;
+      timeframeMs: number;
+    },
+  ): Promise<any[]> {
+    const result = await getHistoricalRates(
+      subgraphUrl,
+      false,
+      filters,
+      undefined,
+      this.rateOracle.id,
+    );
+    return result;
+  }
+
+  public async getHistoricalFixedRate(
+    subgraphUrl: string,
+    filters: {
+      granularity: Granularity;
+      timeframeMs: number;
+    },
+  ): Promise<any[]> {
```

Comment:
`Promise<HistoricalRates[]>` is better

Reply:
[13c3037](https://github.com/Voltz-Protocol/v1-sdk/pull/206/commits/13c30377bd06863b9fa0ffd0afeae77c4d4abda3)